scripts/west_commands/runners/pyocd.py

```diff
@@ -131,16 +139,29 @@ def flash(self, **kwargs):
                 'Cannot flash; no hex ({}) or bin ({}) files found. '.format(
                     self.hex_name, self.bin_name))
 
-        cmd = ([self.pyocd] +
-               ['flash'] +
-               ['-e', 'sector'] +
-               self.flash_addr_args +
-               self.daparg_args +
-               self.target_args +
-               self.board_args +
-               self.frequency_args +
-               self.flash_extra +
-               [fname])
+        if self.tool_opt != '':
+            cmd = ([self.pyocd] +
+                   ['flash'] +
+                   ['-e', 'sector'] +
+                   self.flash_addr_args +
+                   self.daparg_args +
+                   self.target_args +
+                   self.board_args +
+                   self.frequency_args +
+                   [self.tool_opt] +
```

Comment:
Please avoid using copy/paste like this by just making a self.tool_opt_args that is an empty list if there is no --tool-opt, and contains the tool-opt arguments if there was one.